packages/slate-editor/src/extensions/blockquote/BlockquoteExtension.tsx

```diff
@@ -20,6 +20,7 @@ export function BlockquoteExtension(): Extension {
                 BLOCKQUOTE: () => ({ type: QUOTE_NODE_TYPE }),
             }),
         },
+        isRichBlock: isQuoteNode,
```

Comment:
Hm... This is no good. We should not enable `isRichBlock` for blockquotes.

There's additional special formatting for rich blocks appearing in email campaigns. They are constrained to a 600px width column in the email rendering logic. Here's how a right-aligned quote looks now:

![image](https://user-images.githubusercontent.com/370680/186670391-5468026d-e2f6-4cdf-ba8a-ba71c030b408.png)

See #194 and MT-4987

Reply:
I guess we have no other choice, but to introduce a new check callback for this use-case. 

But instead of declaring this globally on extensions, we can pass it as a parameter to the `PasteSlateContentExtension` constructor:

```ts
// PasteSlateContentExtension.ts

/**
 * @param isPreservedBlock Defines which blocks should be preserved, if pasted empty content.
 */
export function PasteSlateContentExtension(
    isPreservedBlock: (editor: Editor, node: Node) => boolean = alwaysFalse,
): Extension {
    return {
        id: EXTENSION_ID,
        withOverrides: withSlatePasting(isPreservedBlock),
    };
}

```

```ts
// getEnabledExtensions.ts

yield PasteSlateContentExtension(
   (editor, node) => isQuoteNode(node) || editor.isRichBlock(node),
);
```

What would you say about this?

Reply:
I thought that Extension params are more about configuration or providing connection to external api. Here i think that removing node when you paste content is a bug and i am not sure that it should be controlled outside

Reply:
> Here i think that removing node when you paste content is a bug

So do you suggest to make this behavior default for all types of nodes?

Reply:
not to all, to those which have empty children (but can have them) and visually not empty. Maybe we should somehow define `isEmpty` behaviour? What do you think, if we have image but have no caption can we consider that this image node is not empty?

Reply:
Sceptical.

While providing this wider `isEmpty` check for images would make sense, it wouldn't make any sense for Quote nodes. As a quote without children is in fact empty. Treating it otherwise, would pose other problems, I am sure.